test/options_test.cc

```diff
@@ -276,6 +277,141 @@ TEST_F(OptionsImplTest, RequestSource) {
   EXPECT_TRUE(util(*(options_from_proto.toCommandLineOptions()), *cmd));
 }
 
+// This test covers --RequestSourcePlugin, which can't be tested at the same time as --RequestSource
+// and some other options. This is the test for the inlineoptionslistplugin.
+TEST_F(OptionsImplTest, InLineOptionsListRequestSourcePlugin) {
+  Envoy::MessageUtil util;
+  const std::string request_source_config =
+      "{"
+      "name:\"nighthawk.in-line-options-list-request-source-plugin\","
+      "typed_config:{"
+      "\"@type\":\"type.googleapis.com/"
+      "nighthawk.request_source.InLineOptionsListRequestSourceConfig\","
+      "options_list:{"
+      "options:[{request_method:\"1\",request_headers:[{header:{key:\"key\",value:\"value\"}}]}]"
+      "},"
+      "}"
+      "}";
+
+  std::unique_ptr<OptionsImpl> options = TestUtility::createOptionsImpl(
+      fmt::format("{} --request-source-plugin-config {} {}", client_name_, request_source_config,
+                  good_test_uri_));
+
+  // Check that our conversion to CommandLineOptionsPtr makes sense.
+  CommandLineOptionsPtr cmd = options->toCommandLineOptions();
```

Comment:
Stylistically, please avoid abbreviations in variable names.

Reply:
Sure thing.